Flaky Testlabel test in searchcache

In an unrelated PR, the testlabel tests in searchcache started failing. Most likely flaky since they are passing on the main brach
```
=== RUN   TestBindExecute_QueryOrTestLabel
    index_filter_test.go:748: 
        	Error Trace:	index_filter_test.go:748
        	Error:      	Not equal: 
        	            	expected: shared.SearchResult{Test:"/d/e/f", LegacyStatus:[]shared.LegacySearchRunResult{shared.LegacySearchRunResult{Passes:1, Total:1}}, Interop:[]int(nil), Subtests:[]string(nil), Diff:shared.TestDiff(nil)}
        	            	actual  : shared.SearchResult{Test:"/a/b/c", LegacyStatus:[]shared.LegacySearchRunResult{shared.LegacySearchRunResult{Passes:1, Total:1}}, Interop:[]int(nil), Subtests:[]string(nil), Diff:shared.TestDiff(nil)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	 (shared.SearchResult) {
        	            	- Test: (string) (len=6) "/d/e/f",
        	            	+ Test: (string) (len=6) "/a/b/c",
        	            	  LegacyStatus: ([]shared.LegacySearchRunResult) (len=1) {
        	Test:       	TestBindExecute_QueryOrTestLabel
    index_filter_test.go:749: 
        	Error Trace:	index_filter_test.go:749
        	Error:      	Not equal: 
        	            	expected: shared.SearchResult{Test:"/a/b/c", LegacyStatus:[]shared.LegacySearchRunResult{shared.LegacySearchRunResult{Passes:1, Total:1}}, Interop:[]int(nil), Subtests:[]string(nil), Diff:shared.TestDiff(nil)}
        	            	actual  : shared.SearchResult{Test:"/d/e/f", LegacyStatus:[]shared.LegacySearchRunResult{shared.LegacySearchRunResult{Passes:1, Total:1}}, Interop:[]int(nil), Subtests:[]string(nil), Diff:shared.TestDiff(nil)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	 (shared.SearchResult) {
        	            	- Test: (string) (len=6) "/a/b/c",
        	            	+ Test: (string) (len=6) "/d/e/f",
        	            	  LegacyStatus: ([]shared.LegacySearchRunResult) (len=1) {
        	Test:       	TestBindExecute_QueryOrTestLabel
--- FAIL: TestBindExecute_QueryOrTestLabel (0.00s)
```

--- a/api/query/cache/index/index_filter_test.go
+++ b/api/query/cache/index/index_filter_test.go
@@ -1,3 +1,4 @@
+//go:build small
 // +build small
 
 // Copyright 2018 The WPT Dashboard Project. All rights reserved.
@@ -722,7 +723,6 @@ func TestBindExecute_QueryOrTestLabel(t *testing.T) {
 	srs, ok := res.([]shared.SearchResult)
 	assert.True(t, ok)
 
-	assert.Equal(t, 2, len(srs))
 	expectedResult := shared.SearchResult{
 		Test: matchingTestName,
 		LegacyStatus: []shared.LegacySearchRunResult{
@@ -744,9 +744,9 @@ func TestBindExecute_QueryOrTestLabel(t *testing.T) {
 			},
 		},
 	}
-
-	assert.Equal(t, expectedResult1, srs[0])
-	assert.Equal(t, expectedResult, srs[1])
+	assert.Equal(t, 2, len(srs))
+	assert.Contains(t, srs, expectedResult)
+	assert.Contains(t, srs, expectedResult1)
 }
 
 func TestBindExecute_TestLabel(t *testing.T) {

